### DISCUSSION TITLE
BORG Database Types - H2, HSQL, Sqlite?, Mysql?

### DISCUSSION THREAD

**Discussion by mikeberger** [mikeberger/borg_calendar#154] [General]
**Sqlite**
BORG 1.10 adds support for Sqlite. To use an Sqlite database, select the Generic JDBC option on the database options tab and enter an sqlite jdbc url, which has the form: jdbc:sqlite:path_to_file. For example: jdbc:sqlite:/C:/Users/i_fle/OneDrive/borg/borg_sqlite.db

I've personally switched from H2 to Sqlite because I like the Sqlite DB Browser.

**H2**
H2 is the default database for BORG and this is fine for 99% of users. One minor concern is that updating to the latest version of H2 (2.0) requires the DB to be deleted and recreated. Since I can't force all BORG users to Export/Import their data, BORG will forever stay on the prior version of H2 (1.4).

**HSQL**
BORG used HSQL as the default before H2. I do not recommend using it for new DBs.

**MYSQL**
BORG 1.10 removes the MySQL-specific options in the UI. This is just to make sure that new users don't try MySQL. Existing MySQL users will just see their Mysql URL show up under the Generic JDBC options tab. All of the other Mysql-specific code remains in BORG, but I personally haven't used MySQL in years. It is no longer tested. Also, changes in Java have made it more difficult to use the MySQL JDBC driver. 

The Sqlite, H2, and HSQL JDBC drivers are packaged with BORG. The MySQL driver is not.

**Comment by RESDXChgfore9hing** [comment]
maybe we can add a new checkbox H2(v2+) ?
so the user can export xml and then click this.
restart borg then import xml.
done,migrated

**Reply by mikeberger** [reply]
The same version of Borg can't use both versions of H2. The H2 class names and jdbc url did not change with the new version. I can't include both the old and new H2 libraries at the same time. I can't release a new version of Borg with the new H2 version that cannot open the old DB format.

There is also no strong reason to go to the new version.

**Reply by RESDXChgfore9hing** [reply]
hmm ok i see.i thought they will have some performance improvement or whatnot.

seems like we need a to manually jar patch it ourselfves then.

**Reply by RESDXChgfore9hing** [reply]
so i assume the full process is.
__
export xml 
replace jars
restart borg
import xml.

**Reply by mikeberger** [reply]
You would also have to edit the classpath in MANIFEST.MF inside borg.jar. 

Or you could checkout the project from github, change the version in model/pom.xml and build it yourself. You would then need to test various functionality as there is a small chance that some SQL statement will be incompatible with the new version of H2.

Assuming it works without errors, I can't imagine that you will notice any differences.
